Tests/Shared.Specs/TypeAssertionSpecs.cs

```diff
@@ -1445,6 +1448,451 @@ public void When_injecting_a_null_predicate_into_TypeSelector_NotBeDecoratedWith
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Expected all types to be in namespace \"DummyNamespace\"" +
+                             " because we want to test the error message," +
+                             " but the following types are in a different namespace:*" +
+                             "*ClassNotInDummyNamespace*" +
+                             "*OtherClassNotInDummyNamespace*.");
+        }
+
+        [Fact]
+        public void When_asserting_a_selection_of_types_in_the_global_namespace_is_in_that_namespace_it_succeeds()
+        {
+            // Arrange
+            var types = new TypeSelector(new[]
+            {
+                typeof(ClassInGlobalNamespace)
+            });
+
+            // Act
+            Action act = () =>
+                types.Should().BeInNamespace(null);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_asserting_a_selection_of_types_in_the_global_namespace_is_in_another_namespace_it_fails()
+        {
+            // Arrange
+            var types = new TypeSelector(new[]
+            {
+                typeof(ClassInGlobalNamespace)
+            });
+
+            // Act
+            Action act = () =>
+                types.Should().BeInNamespace(nameof(DummyNamespace),
+                    "because we want to test the error {0}", "message");
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Expected all types to be in namespace \"DummyNamespace\"" +
+                             " because we want to test the error message," +
+                             " but the following types are in a different namespace:*" +
+                             "*ClassInGlobalNamespace*");
+        }
+
+        [Fact]
+        public void When_asserting_a_selection_of_types_in_a_namespace_is_in_the_global_namespace_it_fails()
+        {
+            // Arrange
+            var types = new TypeSelector(new[]
+            {
+                typeof(ClassInDummyNamespace),
+                typeof(ClassNotInDummyNamespace),
+                typeof(OtherClassNotInDummyNamespace)
+            });
+
+            // Act
+            Action act = () =>
+                types.Should().BeInNamespace(null,
+                    "because we want to test the error {0}", "message");
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Expected all types to be in namespace <null>" +
+                             " because we want to test the error message," +
+                             " but the following types are in a different namespace:*" +
+                             "*ClassInDummyNamespace*" +
+                             "*ClassNotInDummyNamespace*" +
+                             "*OtherClassNotInDummyNamespace*");
+        }
+
+        [Fact]
```

Comment:
Please put the `BeInNamespace` and the `NotBeInNamespace` tests in separate `#regions`s

Reply:
Done!